engine/src/main/java/org/terasology/engine/core/TerasologyEngine.java

```diff
@@ -181,13 +182,16 @@ public TerasologyEngine(TimeSubsystem timeSubsystem, Collection<EngineSubsystem>
 
         // add all subsystem as engine module part. (needs for ECS classes loaded from external subsystems)
         allSubsystems.stream().map(Object::getClass).forEach(this::addToClassesOnClasspathsToAddToEngine);
+
+        // register NUI classes with engine module
+        addToClassesOnClasspathsToAddToEngine(UIWidget.class);
     }
 
     /**
      * Provide ability to set additional engine classpath locations.   This must be called before initialize() or run().
      * @param clazz any class that appears in the resource location to treat as an engine classpath.
      */
-    protected void addToClassesOnClasspathsToAddToEngine(Class<?> clazz) {
+    public void addToClassesOnClasspathsToAddToEngine(Class<?> clazz) {
```

Comment:
FIXME: can revert this permission change now, as I abandoned my attempt to register NUI only for a certain subsystem.